starport/services/networkbuilder/simulation.go

```diff
@@ -0,0 +1,219 @@
+	"github.com/tendermint/starport/starport/pkg/gomodulepath"
+	"github.com/tendermint/starport/starport/pkg/httpstatuschecker"
+	"github.com/tendermint/starport/starport/pkg/xurl"
+	"github.com/tendermint/starport/starport/services/chain"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const ErrValidatorSetNil = "validator set is nil in genesis and still empty after InitChain"
+
+// VerifyProposals generates a genesis file from the current launch information and proposals to verify
+// The function returns false if the generated genesis is invalid
+func (b *Builder) VerifyProposals(ctx context.Context, chainID string, proposals []int, commandOut io.Writer) (bool, error) {
+	chainInfo, err := b.ShowChain(ctx, chainID)
+	if err != nil {
+		return false, err
+	}
+
+	// Get the simulated launch information from these proposals
+	simulatedLaunchInfo, err := b.SimulatedLaunchInformation(ctx, chainID, proposals)
+	if err != nil {
+		return false, err
+	}
+
+	// find out the app's name form url
+	u, err := url.Parse(chainInfo.URL)
+	if err != nil {
+		return false, err
+	}
+	importPath := path.Join(u.Host, u.Path)
+	path, err := gomodulepath.Parse(importPath)
+	if err != nil {
+		return false, err
+	}
+	app := chain.App{
+		ChainID: chainID,
+		Name:    path.Root,
+		Version: cosmosver.Stargate,
+	}
+	chainCmd, err := chain.New(app, true, chain.LogSilent)
+	if err != nil {
+		return false, err
+	}
+
+	// get app dir
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		return false, err
+	}
+	appHome := filepath.Join(homedir, app.ND())
+
+	// create a temporary dir that holds the genesis to test
+	tmpHome, err := ioutil.TempDir("", app.ND() + "*")
+	if err != nil {
+		return false, err
+	}
+	defer os.RemoveAll(tmpHome)
+	err = copy.Copy(appHome, tmpHome)
+	if err != nil {
+		return false, err
+	}
+
+	// generate the genesis to test
+	if err := generateGenesis(ctx, tmpHome, chainInfo, simulatedLaunchInfo, chainCmd); err != nil {
+		fmt.Fprintf(commandOut, "error generating the genesis: %s\n", err.Error())
+		return false, nil
+	}
+
+	// set the config with random ports to test the start command
+	addressAPI, err := setSimulationConfig(tmpHome)
+	if err != nil {
+		return false, err
+	}
+
+	// run validate-genesis command on the generated genesis
+	err = cmdrunner.New().Run(ctx, step.New(
+		step.Exec(
+			app.D(),
+			"validate-genesis",
+			"--home",
+			tmpHome,
+		),
+		step.Stderr(commandOut),	// This is the error of the verifying command, therefore this is the same as stdout
+		step.Stdout(commandOut),
+	))
+	if err != nil {
+		return false, nil
+	}
+
+	// verify that the chain can be started with a valid genesis
+	// run validate-genesis command on the generated genesis
+	ctx, cancel := context.WithTimeout(ctx, time.Minute * 1)
+	exit := make(chan error)
+
+	// Go routine to check the app is listening
+	go func() {
+		defer cancel()
+		exit <- isBlockchainListening(ctx, addressAPI)
```

Comment:
Do we need to start the chain to know if the Genesis is good or is there any other reasons for this?

Reply:
We need to start it to trigger the execution the gentxs and dynamically verify they are good.
`isBlockchainListening` is used so that we now the execution of the gentxs has been performed and there are no error